### ISSUE TITLE
Coroutine inside a CoroutineExceptionHandler causes a java.lang.VerifyError when triggered in Android

### ISSUE THREAD

**Issue by monzee** [Kotlin/kotlinx.coroutines#177]
Given this handler
```kotlin
    val handler = CoroutineExceptionHandler { _, throwable ->
        launch(UI) {
            tv.text = throwable.message
        }
    }
```
when I run this code
```kotlin
        launch(UI + handler) {
            repeat(5) {
                tv.text = "Crashing in ${5 - it}..."
                delay(1_000)
            }
            error("bang.")
        }
```
I get this exception when the error reaches the handler:
```
12-02 18:54:26.635 30276-30276/em.zed.bugreport E/AndroidRuntime: FATAL EXCEPTION: main
                                                                  Process: em.zed.bugreport, PID: 30276
                                                                  java.lang.VerifyError: em/zed/bugreport/CoroExnHandler$$special$$inlined$CoroutineExceptionHandler$1$lambda$1
                                                                      at em.zed.bugreport.CoroExnHandler$$special$$inlined$CoroutineExceptionHandler$1.handleException(CoroutineExceptionHandler.kt:97)
                                                                      at kotlinx.coroutines.experimental.CoroutineExceptionHandlerKt.handleCoroutineException(CoroutineExceptionHandler.kt:38)
                                                                      at kotlinx.coroutines.experimental.StandaloneCoroutine.onCancellation(Builders.kt:176)
                                                                      at kotlinx.coroutines.experimental.JobSupport.completeUpdateState$kotlinx_coroutines_core(Job.kt:685)
                                                                      at kotlinx.coroutines.experimental.JobSupport.updateState$kotlinx_coroutines_core(Job.kt:628)
                                                                      at kotlinx.coroutines.experimental.JobSupport.makeCompleting$kotlinx_coroutines_core(Job.kt:997)
                                                                      at kotlinx.coroutines.experimental.AbstractCoroutine.resumeWithException(AbstractCoroutine.kt:51)
                                                                      at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:53)
                                                                      at kotlinx.coroutines.experimental.ResumeModeKt.resumeMode(ResumeMode.kt:53)
                                                                      at kotlinx.coroutines.experimental.AbstractContinuation.afterCompletion(AbstractContinuation.kt:87)
                                                                      at kotlinx.coroutines.experimental.JobSupport.completeUpdateState$kotlinx_coroutines_core(Job.kt:688)
                                                                      at kotlinx.coroutines.experimental.JobSupport.updateState$kotlinx_coroutines_core(Job.kt:628)
                                                                      at kotlinx.coroutines.experimental.AbstractContinuation.resumeImpl(AbstractContinuation.kt:105)
                                                                      at kotlinx.coroutines.experimental.CancellableContinuationImpl.resumeUndispatched(CancellableContinuation.kt:233)
                                                                      at kotlinx.coroutines.experimental.android.HandlerContext$scheduleResumeAfterDelay$1.run(HandlerContext.kt:56)
                                                                      at android.os.Handler.handleCallback(Handler.java:733)
                                                                      at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                      at android.os.Looper.loop(Looper.java:136)
                                                                      at android.app.ActivityThread.main(ActivityThread.java:5081)
                                                                      at java.lang.reflect.Method.invokeNative(Native Method)
                                                                      at java.lang.reflect.Method.invoke(Method.java:515)
                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:781)
                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:597)
                                                                      at dalvik.system.NativeStart.main(Native Method)
```
It works without the `#launch()` inside the handler but I need to switch to the UI thread to do anything useful in Android. It seems to work fine in a regular JVM project. Using `0.20-alpha-13` but also observed in `0.19.3`.

Workaround: extract the coroutine into a function then call that in the handler.

**Comment by elizarov** [comment]
I cannot reproduce it with a simplified version of code. The culprit seems to be in `tv.text = ...` assignment. What is `tv`?

**Comment by monzee** [comment]
`tv` is a `TextView`. I published the project here: https://github.com/monzee/bugreport . I upgraded to `0.20` and it still happens consistently.

Repro steps:
- launch project
- open action bar menu, hit `Crash me`
- wait 3 seconds

Expected: `bang.` displayed in the `TextView`
What happens: crash dialog.

- Tested on actual devices @ API 24 and API 19
- I don't believe I have set anything special in the project. It's just a plain project generated by AS 3.0.1 with Kotlin enabled then I added the coroutines dependency and the experimental flag.
- In the activity, when I change the [exception handler body](https://github.com/monzee/bugreport/blob/dfb3fd1c2e559db1e624c4f7c56fba17fca21c17/app/src/main/java/em/zed/bugreport/CoroExnHandler.kt#L22) to `workaround(throwable.message)`, it doesn't crash and the expected scenario is observed.
- The exception message I got this time has a bit more info:
> `12-04 21:12:02.899  1296  1296 E AndroidRuntime: java.lang.VerifyError: Verifier rejected class em.zed.bugreport.CoroExnHandler$$special$$inlined$CoroutineExceptionHandler$1$lambda$1 due to bad method kotlin.coroutines.experimental.Continuation em.zed.bugreport.CoroExnHandler$$special$$inlined$CoroutineExceptionHandler$1$lambda$1.create(kotlinx.coroutines.experimental.CoroutineScope, kotlin.coroutines.experimental.Continuation) (declaration of 'em.zed.bugreport.CoroExnHandler$$special$$inlined$CoroutineExceptionHandler$1$lambda$1' appears in /data/app/em.zed.bugreport-2/base.apk)`
